### ISSUE TITLE
SwiftUI?

### ISSUE THREAD

**Issue by anthonylavado** [jellyfin-archive/jellyfin-client-ios#6]
>I was thinking about moving the UI to use SwiftUI which will only work for iOS 13+ but it’ll be cleaner and simpler. I haven’t gotten around to writing it all yet though.

_Originally posted by @r0arkin in https://github.com/jellyfin/jellyfin-client-ios/pull/5#issuecomment-645581874_

**Comment by anthonylavado** [comment]
Obviously this will be dictated by the Jellyfin iOS Roadmap (which is still pending).

Just some points for reference:
* Our current iOS app is https://github.com/jellyfin/jellyfin-expo
 * The Expo app wraps a web view to the interface already on a Jellyfin server and thus takes advantage of design changes there
 * The Expo can run on iOS devices as old as 10.0 or later (though that may not be accurate)
* We should review the iOS Device Support Timeline here: https://en.wikipedia.org/wiki/List_of_iOS_devices#Timeline

**Comment by anthonylavado** [comment]
Without going in to further details at this time - **do it.**

**Comment by kazenshi** [comment]
Is this the repo were the tvOS app would go?

**Comment by anthonylavado** [comment]
@loligans That's correct. Right now it's a basic shell of an iOS app. Login to a Jellyfin server works, and playback for video items works.

**Comment by anthonylavado** [comment]
@r0arkin Let me know if you want to talk about the app's design at any point :-)

**Comment by s3rverro0m** [comment]
> @r0arkin Let me know if you want to talk about the app's design at any point :-)

Definitely! Sorry, I’ve been crazy busy and haven’t gotten around to working on the UI.

**Comment by loligans** [comment]
I’d like to help out with this project. Are there things ready to be worked on?

**Comment by kazenshi** [comment]
@r0arkin @anthonylavado I’d like to help out with this project. Are there things that are ready to be worked on?

**Comment by s3rverro0m** [comment]
> @r0arkin @anthonylavado I’d like to help out with this project. Are there things that are ready to be worked on?

I’m not working on anything specific right now since work has ramped up. 😕

**Comment by anthonylavado** [comment]
@loligans I haven't written up the iOS Client Roadmap yet, but it was discussed internally with some other Jellyfin project team members.

The basic goals here:
- Update the code with iOS 13+ as a target
- Fix API calls to return Video (Movies, TV Shows) and Audio (Music) properly
- Ensure basic playback of common formats
- Ensure communication with the server (playback progress, etc)

After that, we can get in to more things, like design, adding more format support/transcoding requests, adding download support, etc. Right now, it seems to work well enough for most video.

We're on the verge of releasing 10.6 for the server, so our time outside of that has gotten a little more limited. I'm happy to still answer questions though! You can always stop by the chat rooms too :-)
